buckets/redis/bucket_factory.go

```diff
@@ -50,94 +49,51 @@ type bucketFactory struct {
 	redisOpts                 *redis.Options
 	scriptSHA                 string
 	connectionRetries         int
-	flushdbCommand            string
 	connectionNeedsResolution bool
 	numTimesConnResolved      int // For testing and debugging purposes
+
+	// keyMaxIdleTime will be set as the Redis key TTL unless it is overridden by the per bucket
+	// config MaxIdleMillis
+	keyMaxIdleTime time.Duration
 }
 
 // NewBucketFactory creates a new bucketFactory instance.
-// flushdbCommand specifies the name of the Flushdb command. "flushdb"
-// is used if it's empty.
-func NewBucketFactory(redisOpts *redis.Options, connectionRetries int, flushdbCommand string) quotaservice.BucketFactory {
+func NewBucketFactory(redisOpts *redis.Options, connectionRetries int, keyMaxIdleTime time.Duration) quotaservice.BucketFactory {
 	if connectionRetries < 1 {
 		connectionRetries = 1
 	}
-	if flushdbCommand == "" {
-		flushdbCommand = "flushdb"
+
+	if keyMaxIdleTime == 0 {
+		keyMaxIdleTime = 24 * time.Hour
```

Comment:
I'm happy to bike shed on this default time. From what I understand, the cost of a higher time is more memory used by having keys from old version laying around. I'm inclined to decrease it to _minutes_ given that it seems the intent of the quota service is to provide very quick, approximate rate limits over a short period of time.

Reply:
1 day seems fine by me for now.